packages/mobile/src/send/utils.ts

```diff
@@ -0,0 +1,267 @@
+import { ErrorMessages } from 'src/app/ErrorMessages'
+import { FeeType } from 'src/fees/actions'
+import { E164NumberToAddressType, RecipientVerificationStatus } from 'src/identity/reducer'
+import { Recipient, RecipientKind } from 'src/recipients/recipient'
+import {
+  ConfirmationInput,
+  ManuallyValidatedE164NumberToAddress,
+  TransactionData,
+} from 'src/send/reducers'
+import Logger from 'src/utils/Logger'
+
+export const formatDisplayName = (displayName: string) => {
+  if (displayName !== 'Mobile #') {
+    return { displayName, startOfSentenceDisplayName: displayName }
+  }
+
+  return { displayName: 'your contact', startOfSentenceDisplayName: 'Your contract' }
+}
+
+export const getAddressFromPhoneNumber = (
+  e164Number: string,
+  e164NumberToAddress: E164NumberToAddressType,
+  manuallyValidatedE164NumberToAddress: ManuallyValidatedE164NumberToAddress
+): string | null | undefined => {
+  const addresses = e164NumberToAddress[e164Number]
+
+  if (!addresses) {
```

Comment:
Is this the unverified case?

Reply:
Yup, this is either unverified (null value) or unknown (undefined). Unknown should only happen momentarily, if at all, while addresses are being updated.